class2.md

```diff
@@ -0,0 +1,287 @@
+
+It can be helpful to orient yourself by finding out which paths of the tree most data points take. You can assess this on the image above by looking at the `samples` variable shown in each node.
+
+### Feature importance to summarize useful properties in a tree
+
+Feature importance is a commonly used method to summarize the inner workings of a decision tree. It captures how important each feature is for the decision the tree makes. It is always a number between 0 and 1, where 0 means a feature wasn't used at all and 1 means the feature perfectly predicts the target outcome. The feature importance of each feature should always add up to 1.
+
+<p align="center">
+  <img width="600" alt="" src="images/featureImportance.png">
+</p>
+
+The above image summarizes the feature importance for each feature in the tree above.
+
+### Testing a classification algorithm
+
+A confusion matrix (also known as an error matrix) is a way to test the accuracy of your model. A confusion matrix summerizes the results of the classification algorithm by highlighting false positives, false negatives, true positives, and true negatives.
+
+Below is an example of a confusion matrix for a binary classifier.
+
+<p align="center">
+  <img width="400" alt="" src="confusion.png">
+</p>
+
+In this very basic example there are two possible predicted classes: "yes" and "no". We could imagine this classifier might be predicting the presence of a disease where "yes" means that a patient has the disease.
+
+What can we learn from this matrix?
+* This classifier made a total of 165 predictions (e.g., 165 patients were being tested for the presence of that disease).
+* Out of those 165 cases, the classifier predicted "yes" 110 times, and "no" 55 times.
+* In reality, 105 patients in the sample have the disease, and 60 patients do not.
+
+Confusion matrix terminology:
+* true positives (TP): These are cases in which we predicted yes (they have the disease), and they do have the disease.
+* true negatives (TN): We predicted no, and they don't have the disease.
+* false positives (FP): We predicted yes, but they don't actually have the disease. (Also known as a "Type I error.")
+* false negatives (FN): We predicted no, but they actually do have the disease. (Also known as a "Type II error.")
+
+There are many different metrics that can be derived from a confusion matrix. Two of the most basic metrics are *accuracy* and the *error rate* of a classifier.
+* Accuracy: A measure of how often the machine guesses correctly
+  * `(TP+TN)/total = (100+50)/165 = 0.91`
+* Error rate: A measure of how often the machine guesses incorrectly
+  * `(FP+FN)/total = (10+5)/165 = 0.09`
+  * Equivalent to `1 - Accuracy`
+
+You could further interrogate the above problem by evaluating the sensitivity of the algorithm. Sensitivity measures the true positive rate and would show that the algorithm has 100% sensitivity for cat pictures and 0% sensitivity for dog pictures.
+
+### Regression
```

Comment:
The structure of subheadings here doesn't make sense. It's not clear how the regression and linear regression sections fit into the overall structure of the class.